File: src/CSharp/201-300/201.BitwiseANDOfNumbersRange.cs

using System;
using System.Linq;
using System.Text;
using System.Collections.Generic;


namespace PeefyLeetCode.BitwiseANDOfNumbersRange
{
public class Solution {
    public int RangeBitwiseAnd(int m, int n) {
        var count = 0;
        while (m != n)
        {
            m >>= 1;
            n >>= 1;
            count += 1;
        }
        return n << count;
    }
}
}

File: src/Java/201.BitwiseANDOfNumbersRange.java

import java.util.List;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.Comparator;
import java.util.HashMap;
import java.util.Queue;
import java.util.HashSet;
import java.util.LinkedList;


class Solution {
    public int rangeBitwiseAnd(int m, int n) {
        int count = 0; 
        while (m != n){
            m >>= 1;
            n >>= 1;
            count++;
        }
        return n << count;
    }
}
